.github/scripts/checkLiquidation.mjs

```diff
@@ -0,0 +1,157 @@
+};
+
+const expectedLiquidating = {
+  ids: [
+    'published.vaultFactory.managers.manager0.vaults.vault10-liquidating',
+    'published.vaultFactory.managers.manager0.vaults.vault11-liquidating',
+    'published.vaultFactory.managers.manager0.vaults.vault12-liquidating',
+  ],
+  debts: ['100500000', '103515000', '105525000'],
+  balance: ['15000000', '15000000', '15000000'],
+  denom: 'ATOM',
+};
+
+const expectedLiquidated = {
+  ids: [
+    'published.vaultFactory.managers.manager0.vaults.vault10-liquidated',
+    'published.vaultFactory.managers.manager0.vaults.vault11-liquidated',
+    'published.vaultFactory.managers.manager0.vaults.vault12-liquidated',
+  ],
+  debts: ['0', '0', '0'],
+  balance: ['3425146', '3077900', '2846403'],
+  denom: 'ATOM',
+};
+
+const validate = async ({
+  apiUrl,
+  maxRetries,
+  retryDuration,
+  expectedIds,
+  expectedDebts,
+  expectedBalance,
+  expectedDenom,
+  filterState,
+}) => {
+  console.log(`Checking ${filterState} vaults...`);
+  let retries = 0;
+
+  while (retries < maxRetries) {
+    try {
+      const graphqlQuery = {
+        query: `query {
+            vaultLiquidations (filter: {state: {equalTo: "${filterState}"}}) {
+                nodes {
+                    id
+                    denom
+                    debt
+                    state
+                    balance
+                }
+            }
+        }`,
+      };
+
+      const response = await fetch(apiUrl, {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+          Accept: 'application/json',
+        },
+        body: JSON.stringify(graphqlQuery),
+      });
+
+      const jsonResponse = await response.json();
+      console.log('Response:', JSON.stringify(jsonResponse));
+
+      const nodes = jsonResponse?.data?.vaultLiquidations?.nodes || [];
+      if (nodes.length !== 3) {
+        console.error(`Attempt ${retries + 1}: No data available`);
+        retries++;
+        await delay(retryDuration);
+        continue;
+      }
+
+      nodes.sort((a, b) => a.id.localeCompare(b.id));
+      console.log('Sorted Nodes:', JSON.stringify(nodes));
+
+      // Validate Ids
+      assert.strictEqual(nodes[0].id, expectedIds[0]);
+      assert.strictEqual(nodes[1].id, expectedIds[1]);
+      assert.strictEqual(nodes[2].id, expectedIds[2]);
+
+      // Validate Debts
+      assert.strictEqual(nodes[0].debt, expectedDebts[0]);
+      assert.strictEqual(nodes[1].debt, expectedDebts[1]);
+      assert.strictEqual(nodes[2].debt, expectedDebts[2]);
+
+      // Validate Denom
+      assert.strictEqual(nodes[0].denom, expectedDenom);
+      assert.strictEqual(nodes[1].denom, expectedDenom);
+      assert.strictEqual(nodes[2].denom, expectedDenom);
+
+      // Validate Balance
+      assert.strictEqual(nodes[0].balance, expectedBalance[0]);
+      assert.strictEqual(nodes[1].balance, expectedBalance[1]);
+      assert.strictEqual(nodes[2].balance, expectedBalance[2]);
```

Comment:
by importing the strict module above, you don't have to repeat it here,
```suggestion
      assert.equal(nodes[0].id, expectedIds[0]);
      assert.equal(nodes[1].id, expectedIds[1]);
      assert.equal(nodes[2].id, expectedIds[2]);
```

but these lines are pretty verbose and you have to update them if you change the nodes or expectations. once you've made the change to take a single `expectations` parameter you can do,

```suggestion
      for (const key of Object.keys(expectations) {
        for (let i = 0; i < nodes.length; i++) {
          assert.equal(nodes[i][key], expecations[i].key);
        }
      }
```

Oh though you'll also have to take the plural `s` off the expectation fixture.